src/ws2st/forces/getset_property.cpp

```diff
@@ -0,0 +1,141 @@
+            std::cout << "\tReplaced a call to get the " << kindString << " property [[" << name << "]]" << std::endl;
+        }
+    }
+
+    StatementMatcher SetPosMatcher = PropertySetMatcher("set_pos");
+    StatementMatcher SetVelMatcher = PropertySetMatcher("set_vel");
+    StatementMatcher SetAccMatcher = PropertySetMatcher("set_acc");
+
+    StatementMatcher SetDensityMatcher = PropertySetMatcher("set_density");
+    StatementMatcher SetMassMatcher = PropertySetMatcher("set_mass");
+    StatementMatcher SetSmoothingLengthMatcher = PropertySetMatcher("set_smoothing_length");
+
+    WashCallbackFn HandleSetPos = &HandleSetProperty<ForceType::VECTOR, PropertyList::Pos>;
+    WashCallbackFn HandleSetVel = &HandleSetProperty<ForceType::VECTOR, PropertyList::Vel>;
+    WashCallbackFn HandleSetAcc = &HandleSetProperty<ForceType::VECTOR, PropertyList::Acc>;
+
+    WashCallbackFn HandleSetDensity = &HandleSetProperty<ForceType::SCALAR, PropertyList::Density>;
+    WashCallbackFn HandleSetMass = &HandleSetProperty<ForceType::SCALAR, PropertyList::Mass>;
+    WashCallbackFn HandleSetSmoothingLength = &HandleSetProperty<ForceType::SCALAR, PropertyList::SmoothingLength>;
+
+    template <ForceType type, PropertyList property>
+    void HandleSetProperty(const MatchFinder::MatchResult& Result, Replacements& Replace)  {
+        const auto *call = Result.Nodes.getNodeAs<CXXMemberCallExpr>("callExpr");
+        const auto setValue = Result.Nodes.getNodeAs<Expr>("setValue");
+        const Expr *objectExpr = call->getImplicitObjectArgument();
+
+        if (!call) {
+            std::cout << "Matched a node with no call found" << std::endl;
+            throw std::runtime_error("Matched a set property call with no call node.");
+        }
+
+        std::string name = propertyName(property);
+
+        constexpr const char *kindString = (type == ForceType::SCALAR) ? "scalar" : "vector";
+        std::cout << "Picked up " << kindString << " "
+                    << getSourceText(Result.Context, call->getSourceRange()).value() << std::endl;
+        std::string objectCodeStr = getSourceText(Result.Context, objectExpr->getSourceRange()).value();
+        std::string setValueStr = getSourceText(Result.Context, setValue->getSourceRange()).value();
+
+        std::string replacementStr = "(wash::" + (std::string)kindString + "_force_" + name + ")[" +
+                                        objectCodeStr + ".get_id()] = " + setValueStr;
+
+        auto Err = Replace.add(Replacement(
+            *Result.SourceManager, CharSourceRange::getTokenRange(call->getSourceRange()), replacementStr));
+
+        if (Err) {
+            std::cout << llvm::toString(std::move(Err)) << std::endl;
+        } else {
+            std::cout << "\tReplaced a call to set the " << kindString << " property [[" << name << "]]" << std::endl;
+        }
+    }
+
+    const std::string propertyName(PropertyList property) {
+        if (property == PropertyList::Pos) {
```

Comment:
Could do with a `switch` statement here. 

Also, are we okay to hardcode these properties here?

Reply:
Yeah. These properties are the pre-defined on the particle class. Always present in the simulation. Cause they're accessed through the methods like `get_vel` or `set_density` becomes necessary to define these rules slightly differently.

Reply:
Changed to switch stmt